Core/include/Acts/MagneticField/InterpolatedBFieldMap.hpp

```diff
@@ -278,47 +264,35 @@ class InterpolatedBFieldMap : public InterpolatedMagneticField {
   ///
   /// @pre The given @c position must lie within the range of the underlying
   ///      magnetic field map.
-  Result<Vector3> getField(const Vector3& position) const {
-    const auto gridPosition = m_cfg.transformPos(position);
-    if (!isInsideLocal(gridPosition)) {
-      return Result<Vector3>::failure(MagneticFieldError::OutOfBounds);
-    }
-
-    return Result<Vector3>::success(
-        m_cfg.transformBField(m_cfg.grid.interpolate(gridPosition), position));
-  }
-
-  Vector3 getFieldUnchecked(const Vector3& position) const final {
+  Vector3 getField(const Vector3& position) const {
     const auto gridPosition = m_cfg.transformPos(position);
     return m_cfg.transformBField(m_cfg.grid.interpolate(gridPosition),
                                  position);
```

Comment:
I think relying on the over/underflow behavior of the grid here is dangerous. I think the chance that the user will intentionally and correctly initialize this is very small, with the overwhelming majority of cases being silent random magnetic field values.

Reply:
but that overflow bins are set can be asserted on when constructing the field no?

Reply:
I don't think we should be second-guessing the users: if the service can return a well-defined B-field, then return that. We have a mechanism for under/overflow, so we should apply it, not give an error. I think it is reasonable to give an error if the under/overflow B-field are not filled. Certainly they should not be "random magnetic field values".

Is there a problem with the client code if we return `B=(0,0,0)`? If not, I personally think that would be a reasonable default even for unspecified bins. If outside the range of the magnet, we have zero (or residual) field. I realise that may be a minority view, so it's fine to require that all bins be specified.

Reply:
There's no way to opt out of the under/overflow behavior if we remove the error mechanism.